alibuild_helpers/utilities.py

```diff
@@ -132,7 +132,11 @@ def format(s, **kwds):
 
 def doDetectArch(hasOsRelease, osReleaseLines, platformTuple, platformSystem, platformProcessor):
   if platformSystem == "Darwin":
-    return "osx_x86-64"
+    systname,nodename,release,version,machine = os.uname()
```

Comment:
Hm, this seems to make the unit tests fail. Would it be possible to prefer `platformProcessor` over what `os.uname()` returns? i.e.
```python
machine = platformProcessor or os.uname()[4]
if machine == ...
```